src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernelUtilities.hpp

```diff
@@ -1221,12 +1249,19 @@ struct computePotentialGravity
     }
 
     //inner loop to get average density
+    integer denom = 0;
     for( localIndex i = 0; i < numFluxSupportPoints; ++i )
     {
       localIndex const er = seri[i];
       localIndex const esr = sesri[i];
       localIndex const ei = sei[i];
 
+      bool const phaseExists = (phaseVolFrac[er][esr][ei][ip] > 0);
```

Comment:
In some cases I have seen saturation weighted density in this case. Not sure what the benefit of that is over a simpler mean. Also looks like it might complicate the derivatives.

Reply:
Yeah, it should be more accurate, but I don't want to complicate the derivatives